File: HurpApp/app/src/main/java/app/recrutamento/android/challengealpha/model/hotel/Filters.kt
package app.recrutamento.android.challengealpha.model.hotel

import com.google.gson.annotations.Expose
import com.google.gson.annotations.SerializedName

class Filters {

    @SerializedName("amenities")
    @Expose
    var amenities: List<Amenity>? = null
    @SerializedName("attributes")
    @Expose
    var attributes: List<Attribute>? = null
    @SerializedName("countries")
    @Expose
    var countries: List<Country>? = null
    @SerializedName("cities")
    @Expose
    var cities: List<City>? = null
    @SerializedName("departureCities")
    @Expose
    var departureCities: List<DepartureCity>? = null
    @SerializedName("prices")
    @Expose
    var prices: List<Price>? = null
    @SerializedName("priceInterval")
    @Expose
    var priceInterval: PriceInterval? = null
    @SerializedName("productType")
    @Expose
    var productType: List<ProductType>? = null
    @SerializedName("regulation")
    @Expose
    var regulation: List<Regulation>? = null
    @SerializedName("rooms")
    @Expose
    var rooms: List<Room>? = null
    @SerializedName("stars")
    @Expose
    var stars: List<Star>? = null
    @SerializedName("states")
    @Expose
    var states: List<State>? = null

}

File: DesafioHurb/Domain/Model/Filters.swift
//
//  Filters.swift
//  DesafioHurb
//
//  Created by Edson Aparecido Guido on 13/09/20.
//  Copyright © 2020 Edson Aparecido Guido. All rights reserved.
//

import Foundation

struct Filters: Codable, Equatable {
    let amenities: [Filter]
    let attributes: [Filter]
    let countries: [Filter]
    let cities: [Filter]
    let departureCities: [Filter]
    let duration: [Filter]
    let food: [Filter]
    let people: [Filter]
    let prices: [PriceElement]
    let priceInterval: PriceInterval
    let productType: [Filter]
    let regulation: [Filter]
    let rooms: [Filter]
    let stars: [Filter]
    let states: [Filter]
}
